l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -228,16 +234,21 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
     /// @dev for backwards compatibility and to automatically register l1 assets, and we return the correct info.
     function _getAssetProperties(bytes32 _assetInfo) internal returns (address l1Asset, bytes32 assetInfo) {
         l1Asset = assetAddress[_assetInfo];
-        assetInfo = _assetInfo;
         if (l1Asset == address(0) && (uint256(_assetInfo) <= type(uint160).max)) {
             l1Asset = address(nativeTokenVault);
             assetInfo = keccak256(abi.encode(block.chainid, NATIVE_TOKEN_VAULT_VIRTUAL_ADDRESS, _assetInfo));
+            assetAddress[_assetInfo] = l1Asset;
+            assetAddress[assetInfo] = l1Asset;
             nativeTokenVault.registerToken(address(uint160(uint256(_assetInfo))));
+        } else if (uint256(_assetInfo) <= type(uint160).max) {
+            assetInfo = keccak256(abi.encode(block.chainid, NATIVE_TOKEN_VAULT_VIRTUAL_ADDRESS, _assetInfo));
```

Comment:
Can you make a call to the NTV here instead? Less efficient but more secure probably

Reply:
updated